src/test/regress/sql/multi_shard_update_delete.sql

```diff
@@ -0,0 +1,271 @@
+SELECT create_reference_table('events_reference_copy_table');
+INSERT INTO events_reference_copy_table SELECT * FROM events_reference_table;
+
+CREATE TABLE users_reference_copy_table (like users_reference_table INCLUDING DEFAULTS INCLUDING CONSTRAINTS INCLUDING INDEXES);
+SELECT create_reference_table('users_reference_copy_table');
+INSERT INTO users_reference_copy_table SELECT * FROM users_reference_table;
+
+-- Run multi shard updates and deletes without transaction on hash distributed tables
+SELECT COUNT(*) FROM users_copy_table;
+UPDATE users_copy_table SET value_1 = 1;
+SELECT SUM(value_1) FROM users_copy_table;
+
+SELECT COUNT(*) FROM users_copy_table WHERE user_id = 20 or user_id = 8 or user_id = 60;
+SELECT SUM(value_2) FROM users_copy_table WHERE user_id = 20 or user_id = 8 or user_id = 60;
+UPDATE users_copy_table SET value_2 = value_2 + 1 WHERE user_id = 20 or user_id = 8 or user_id = 60;
+SELECT SUM(value_2) FROM users_copy_table WHERE user_id = 20 or user_id = 8 or user_id = 60;
+
+UPDATE users_copy_table SET value_3 = 0 WHERE user_id <> 20;
+SELECT SUM(value_3) FROM users_copy_table;
+
+SELECT COUNT(*) FROM users_copy_table WHERE user_id = 20 or user_id = 8 or user_id = 60;
+DELETE FROM users_copy_table WHERE user_id = 20 or user_id = 8 or user_id = 60;
+SELECT COUNT(*) FROM users_copy_table WHERE user_id = 20 or user_id = 8 or user_id = 60;
+
+-- Run multi shard update delete queries within transactions
+BEGIN;
+UPDATE users_copy_table SET value_3 = 0;
+END;
+SELECT SUM(value_3) FROM users_copy_table;
+
+-- Update can also be rollbacked
+BEGIN;
+UPDATE users_copy_table SET value_3 = 1;
+ROLLBACK;
+SELECT SUM(value_3) FROM users_copy_table;
+
+-- Run with inserts (we need to set citus.multi_shard_modify_executor to sequential)
+BEGIN;
+INSERT INTO users_copy_table (user_id, value_1) VALUES(20, 15);
+INSERT INTO users_copy_table (user_id, value_1) VALUES(16,1), (20,16), (7,1), (20,17);
+SET citus.multi_shard_modify_executor to sequential;
+UPDATE users_copy_table SET value_3 = 1;
+END;
+SELECT SUM(value_3) FROM users_copy_table;
+
+-- Run multiple multi shard updates (with sequential executor)
+BEGIN;
+UPDATE users_copy_table SET value_3 = 5;
+UPDATE users_copy_table SET value_3 = 0;
+END;
+SELECT SUM(value_3) FROM users_copy_table;
+
+-- Run multiple multi shard updates (with parallel executor)
+SET citus.multi_shard_modify_executor to DEFAULT;
+UPDATE users_copy_table SET value_3 = 5;
+BEGIN;
+UPDATE users_copy_table SET value_3 = 2;
+UPDATE users_copy_table SET value_3 = 0;
+END;
+SELECT SUM(value_3) FROM users_copy_table;
+
+-- Check with kind of constraints
+UPDATE users_copy_table SET value_3 = 1 WHERE user_id = 20 or true;
+SELECT SUM(value_3) FROM users_copy_table;
+UPDATE users_copy_table SET value_3 = 0 WHERE user_id = 20 and false;
+SELECT SUM(value_3) FROM users_copy_table;
+
+-- Run multi shard updates with prepared statements
+PREPARE foo_plan(int,int) AS UPDATE users_copy_table SET value_1 = $1, value_3 = $2;
```

Comment:
is there a way to verify that we're not accidentally running a multi-shard update when we have a prepared statement with `WHERE user_id = $1`?